test/test-TabTracker.js

```diff
@@ -359,13 +359,29 @@ exports.test_TabTracker_action_pings = function*(assert) {
   for (let key of additionalKeys) {
     assert.ok(pingData[key], `The ping has the additional key ${key}`);
   }
-  assert.deepEqual(eventData.msg.data, pingData, "We receive the expected ping data.");
+  assert.deepEqual(eventData.msg.data.source, pingData.source, "the ping has the correct source");
+  assert.deepEqual(eventData.msg.data.event, pingData.event, "the ping has the correct event");
+  assert.deepEqual(eventData.msg.data.action_position, pingData.action_position, "the ping has the correct action_position");
 };
 
 exports.test_TabTracker_unload_reason_with_user_action = function*(assert) {
```

Comment:
In this test, I added promises to wait for tab open and close. This is because `session_id` will only be set after a tab is open (and we must close tabs at the end of every test).